packages/pwa-kit-create-app/scripts/create-mobify-app.js

```diff
@@ -186,9 +188,9 @@ const runGenerator = (answers, {outputDir}) => {
 }
 
 const retailReactAppPrompts = () => {
-    const validProjectId = (s) =>
-        /^[a-z0-9-]{1,20}$/.test(s) ||
-        'Value can only contain lowercase letters, numbers, and hyphens.'
+    const validProjectName = (s) =>
+        /^[a-zA-Z0-9-\s]{1,20}$/.test(s) ||
```

Comment:
We reuse this regex to validate the project name in the hello world prompts as well. We could extract it so we're not repeating it multiple times for each template prompt.

Also, we may want to rewrite this so we're making use of the PROJECT_ID_MAX_LENGTH constant that was added.
Something like: 
new RegExp(`^[a-zA-Z0-9-\\s]{1,${PROJECT_ID_MAX_LENGTH}\$`)

Reply:
TBH, I also thought about re-using the constant inside the Regex, but I discarded the idea.

Re-use a constant dynamically inside a Regex implies making this one regex different from the rest of the regex used in the generator.
It needs to use the Regex() constructor with template literals and escape special symbols. To me, this optimization looks more difficult to read. But I applied the change. Let me know your thoughts.

Reply:
Good point about the regex readability becoming more difficult because of the template literal escape symbols. Now you've got me leaning towards the original approach as well. 

I think, regardless of the approach we take, we probably want to add a comment either saying what the regex is meant to be outside of the template literal or a warning that the regex needs to be updated if PROJECT_ID_MAX_LENGTH is modified.

I like the idea of extracting the validation we reuse into a function.

Reply:
I propose to leave it as it is now because I don't think that adding code with the original regex in the form of a comment is going to be very helpful. 
The compiler ignores comments, so the regex code added in the comment can easily get obsolete and contradict what the code does in future changes.